src/dependency-manager/src/manager.ts

```diff
@@ -451,8 +454,17 @@ export default abstract class DependencyManager {
     const environment_components = interpolated_environment.getComponents();
     // Prefix interpolation expressions with components.<name>.
     const prefixed_component_map: Dictionary<ComponentConfig> = {};
+
+    // pre-sort values dictionary to properly stack/override any colliding keys
+    const sorted_values_keys = Object.keys(this.values_dictionary).sort();
```

Comment:
Can you add a new values.ts file to the dependency manager that has a validate function? Make the method take the values json so we can use it api side as well. Then call the validate method in the dep manager. 

Things to validate:
1. Structure - double nested
2. Keys only have a wildcard at the end of the string
3. ?

Reply:
I added that class and added tests, I'm not too sure where we'd like this to be called from though. I'm not sure that calling it in `init` feels right, but it's also the first place that the file hits the manager, so that does seem right. I'm a little torn. Also the fact that we need the original file contents in order to produce line numbers for validation errors makes me wonder if we should run the validation in the local manager instead. Seems strange if we were to pass file contents to the base manager